router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -38,23 +39,27 @@ func init() {
 	pkgLogger = logger.NewLogger().Child("asyncdestinationmanager").Child("marketobulkupload")
 }
 
-func (b *MarketoBulkUploader) Poll(importingJob *jobsdb.JobT, payload []byte, timeout time.Duration) ([]byte, int) {
+func (b *MarketoBulkUploader) Poll(pollStruct common.AsyncPoll) ([]byte, int) {
+	payload, err := json.Marshal(pollStruct)
+	if err != nil {
+		panic("JSON Marshal Failed" + err.Error())
+	}
 	transformUrl := config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
-	parameters := importingJob.LastJobStatus.Parameters
-	pollUrl := gjson.GetBytes(parameters, "pollURL").String()
-	bodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(transformUrl+pollUrl, payload, timeout)
+	pollUrl := "/pollStatus"
+	bodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(transformUrl+pollUrl, payload, b.timeout)
 	return bodyBytes, statusCode
 }
 
-func (b *MarketoBulkUploader) FetchFailedEvents(destination *backendconfig.DestinationT, destStruct *utils.DestinationWithSources, importingList []*jobsdb.JobT, importingJob *jobsdb.JobT, asyncResponse common.AsyncStatusResponse, timeout time.Duration) ([]byte, int) {
+func (b *MarketoBulkUploader) FetchFailedEvents(destStruct *utils.DestinationWithSources, importingList []*jobsdb.JobT, importingJob *jobsdb.JobT, asyncResponse common.AsyncStatusResponse) ([]byte, int) {
 	transformUrl := config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
 	failedJobUrl := asyncResponse.FailedJobsURL
 	parameters := importingJob.LastJobStatus.Parameters
 	importId := gjson.GetBytes(parameters, "importId").String()
 	csvHeaders := gjson.GetBytes(parameters, "metadata.csvHeader").String()
-	destType := destination.DestinationDefinition.Name
-	payload := common.GenerateFailedPayload(destStruct.Destination.Config, importingList, importId, destType, csvHeaders)
-	failedBodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(transformUrl+failedJobUrl, payload, common.HTTPTimeout)
+	payload := common.GenerateFailedPayload(destStruct.Destination.Config, importingList, importId, b.destName, csvHeaders)
+	//payload eg: "{\"config\":{\"clientId\":\"01a70f1f-ff37-46fc-bdff-42e92a3f2bb3\",\"clientSecret\":\"rziQBHtZ34Vl1CE3x3OiA3n8Wr45lwar\",\"columnFieldsMapping\":[{\"from\":\"anonymousId\",\"to\":\"anonymousId\"},{\"from\":\"address\",\"to\":\"address\"},{\"from\":\"email\",\"to\":\"email\"}],\"deDuplicationField\":\"\",\"munchkinId\":\"585-AXP-425\",\"oneTrustCookieCategories\":[],\"uploadInterval\":\"10\"},\"input\":[{\"message\":{\"address\":23,\"anonymousId\":\"Test Kinesis\",\"email\":\"test@kinesis.com\"},\"metadata\":{\"job_id\":5}}],\"destType\":\"marketo_bulk_upload\",\"importId\":\"3095\""
```

Comment:
put is indented format for better readability, same in all other places and make sure it doesn't have any sensitive information

Reply:
should we keep these? I added this for ease of understanding